src/teleop/gui/src/components/Chlorophyll.vue

```diff
@@ -0,0 +1,184 @@
+                <ToggleButton id="uv_carousel" v-bind:currentState="UV_carousel" labelEnableText="Carousel UV On" labelDisableText="Carousel UV Off" v-on:change="toggleCarouselUV()"/>
+            </div>
+        </div>
+        <div class="wrap-table">
+            <div>
+                <h3> Chlorophyll Spectral data </h3>
+            </div>
+            <table class="tableFormat" style="undefined;table-layout: fixed; width: 434px">
+                <colgroup>
+                    <col style="width: 63px">
+                    <col style="width: 53px">
+                    <col style="width: 53px">
+                    <col style="width: 53px">
+                    <col style="width: 53px">
+                    <col style="width: 53px">
+                    <col style="width: 53px">
+                    <col style="width: 53px">
+                </colgroup>
+                <thead>
+                <tr>
+                    <th class = "tableElement"></th>
+                    <th class = "tableElement">1</th>
+                    <th class = "tableElement">2</th>
+                    <th class = "tableElement">3</th>
+                    <th class = "tableElement">4</th>
+                    <th class = "tableElement">5</th>
+                    <th class = "tableElement">6</th>
+                </tr>
+                </thead>
+                <tbody>
+                    <tr>
+                        <td class = "tableElement">Spec 0</td>
+                        <td class = "tableElement" v-for="i in 6" :key="i">{{ (spectral_data[i-1]).toFixed(0) }}</td>
+                    </tr>
+                </tbody>
+            </table>
+        </div>
+        <GenerateReport v-bind:spectral_data="spectral_data"/>
+    </div>
+</template>
+
+<script>
+import ToggleButton from './ToggleButton.vue';
+import ROSLIB from 'roslib';
+import GenerateReport from './GenerateReport.vue';
+
+export default {
+    data() {
+        return {
+            whiteLEDS_active: false,
+            UV_endEffector: false,
+            UV_carousel: false
+        }
+    },
+
+    props: {
+        spectral_data: {
+            type: Array,
+            required: true
+        }
+    },
+
+    components: {
+        ToggleButton,
+        GenerateReport
+    },
+
+    methods: {
+        toggle_whiteLEDS: function () {
+            this.whiteLEDS_active = !this.whiteLEDS_active
+            let whiteLedService = new ROSLIB.Service({
+                ros : this.$ros,
+                name : 'change_white_led_state',
+                serviceType : 'mrover/ChangeDeviceState'
+            });
+            let request = new ROSLIB.ServiceRequest({
+                enable: this.whiteLEDS_active
+            });
+            whiteLedService.callService(request, (result) => {
+                if (!result) {
+                    this.whiteLEDS_active = !this.whiteLEDS_active
```

Comment:
idk if this boolean is particularly true... i guess it would make it more obvious to the user that something is wrong though.